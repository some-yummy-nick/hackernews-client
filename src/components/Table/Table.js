import React from "react";
import { Button } from "../Button/Button";
import './Table.css';

export const Table = ({ list, onDismiss }) =>
	<div className="table">
	{list.map(item =>
			<div key={item.objectID} className="table-row">
						<span>
						<a href={item.url}>{item.title}</a>
						</span>
				<span>{" "}{item.author}</span>
				<span>{" "}{item.num_comments}</span>
				<span>{" "}{item.points}{" "}</span>
				<span>
						<Button onClick={() => onDismiss(item.objectID)} className="button-inline">
							Отбросить
						</Button>
						</span>
			</div>
		)}
	</div>;

