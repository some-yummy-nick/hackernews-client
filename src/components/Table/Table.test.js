import React from "react";
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import {Table} from "./Table";

describe('Table', () => {
	const props = {
		list: [
			{ title: 'title 1', author: 'name 1', num_comments: 1, points: 2, objectID: 'y' },
			{ title: 'title 2', author: 'name 2', num_comments: 1, points: 2, objectID: 'z' },
		],
	};
	test('отрисовывает без ошибки', () => {
		const div = document.createElement('div');
		ReactDOM.render(<Table { ...props } />, div);
	});
	test('есть корректный снимок', () => {
		const component = renderer.create(
			<Table { ...props } />
		);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
