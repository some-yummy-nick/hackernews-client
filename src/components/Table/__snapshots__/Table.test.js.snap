// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table есть корректный снимок 1`] = `
<div
  className="table"
>
  <div
    className="table-row"
  >
    <span>
      <a>
        title 1
      </a>
    </span>
    <span>
       
      name 1
    </span>
    <span>
       
      1
    </span>
    <span>
       
      2
       
    </span>
    <span>
      <button
        className="button-inline"
        onClick={[Function]}
        type="button"
      >
        Отбросить
      </button>
    </span>
  </div>
  <div
    className="table-row"
  >
    <span>
      <a>
        title 2
      </a>
    </span>
    <span>
       
      name 2
    </span>
    <span>
       
      1
    </span>
    <span>
       
      2
       
    </span>
    <span>
      <button
        className="button-inline"
        onClick={[Function]}
        type="button"
      >
        Отбросить
      </button>
    </span>
  </div>
</div>
`;
